Gen3UI/VideoStream/JniNative.cpp

```diff
@@ -0,0 +1,73 @@
+#include "JniNative.h"
+#include <QMutex>
+QMutex g_notifyMutex;
+bool g_b_notify;
+int g_msgNo, g_x, g_y;
+#if defined(ANDROID)
+//javah -jni java.util.Date
+//javap -s -p java.util.Date
+
+static void notifyMsg(JNIEnv *env, jobject thiz, int msgNo, int x, int y)
+{
+    g_notifyMutex.lock();
+    g_b_notify = true;
+    g_msgNo = msgNo;
+    g_x = x;
+    g_y = y;
+    g_notifyMutex.unlock();
+}
+
+uchar * gBuffer;
+static void setDirectBuffer(JNIEnv *env, jobject thiz, jobject buffer, jint len)
+{
+    //无需拷贝，直接获取与Java端共享的直接内存地址(效率比较高，但object的构造析构开销大，建议长期使用的大型buffer采用这种方式)
+    gBuffer = (unsigned char *)env->GetDirectBufferAddress(buffer);
+    if( gBuffer == NULL ) {
+        QDBG<<"GetDirectBufferAddress Failed!";
+        return;
+    }
+    else
+    {
```

Comment:
Please reference below format:
```
// Curly braces around both IF and ELSE required because
// one of the clauses used braces.
if (condition) {
  foo;
} else {
  bar;
}
```